boot/util.ml

```diff
@@ -48,7 +48,7 @@ let use_suffix prefix suffix =
   then suffix
   else Filename.concat prefix suffix
 
-let canonical_path_name p =
+let canonical_dir p =
```

Comment:
Please don't do these kind of changes in unrelated PRs.

Due to the renaming extra noise, it took me much longer to review this PR, and I couldn't do as much of as precise job.

Reply:
Sorry about that, I should have told that the renaming is in a separate commit.

More generally, even with the renaming, I'm not fully satisfied by the terminology used in `CUnix`. We should fix some naming conventions at some time.